packages/react-reconciler/src/ReactFiberHydrationContext.js

```diff
@@ -577,6 +672,23 @@ function prepareToHydrateHostSuspenseInstance(fiber: Fiber): void {
   hydrateSuspenseInstance(suspenseInstance, fiber);
 }
 
+function skipPastDehydratedActivityInstance(
+  fiber: Fiber,
+): null | HydratableInstance {
+  const activityState: null | ActivityState = fiber.memoizedState;
+  const activityInstance: null | ActivityInstance =
+    activityState !== null ? activityState.dehydrated : null;
+
+  if (!activityInstance) {
+    throw new Error(
+      'Expected to have a hydrated suspense instance. ' +
+        'This error is likely caused by a bug in React. Please file an issue.',
+    );
+  }
+
+  return getNextHydratableInstanceAfterActivityInstance(activityInstance);
```

Comment:
>_possible bug_

**✨ AI Suggestion:** The `activityInstance` variable might be null. Add a null check before using it.
```suggestion
  const activityState: null | ActivityState = fiber.memoizedState;
  const activityInstance: null | ActivityInstance =
    activityState !== null ? activityState.dehydrated : null;

  if (activityInstance !== null) {
    return getNextHydratableInstanceAfterActivityInstance(activityInstance);
  } else {
    throw new Error(
      'Expected to have a hydrated activity instance. ' +
        'This error is likely caused by a bug in React. Please file an issue.',
    );
  }
```